src/contracts/Output/VisitorAdapterNormalizer.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Rest\Output;
+
+use Ibexa\Contracts\Rest\Output\Generator as BaseGenerator;
+use Ibexa\Rest\Output\Generator;
+use LogicException;
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class VisitorAdapterNormalizer implements NormalizerInterface, NormalizerAwareInterface, ContextAwareNormalizerInterface
+{
+    use NormalizerAwareTrait;
+
+    private const string CALLED_CONTEXT = __CLASS__ . '_CALLED';
+
+    public const string ENCODER_CONTEXT = 'ENCODER_CONTEXT';
+
+    public function __construct(
+        private readonly EncoderInterface $jsonEncoder,
+        private readonly EncoderInterface $xmlEncoder,
+        private readonly ValueObjectVisitorResolverInterface $valueObjectVisitorResolver,
+    ) {
+    }
+
+    /**
+     * @param array<string, mixed> $context
+     */
```

Comment:
CS
```suggestion
    /**
     * @param array<string, mixed> $context
     *
     * @throws \Symfony\Component\Serializer\Exception\ExceptionInterface
     */
```

Reply:
As Paweł mentioned, this exception is part of the `NormalizerInterface`.